homeassistant/components/neato/config_flow.py

```diff
@@ -1,112 +1,52 @@
-            step_id="user",
-            data_schema=vol.Schema(
-                {
-                    vol.Required(CONF_USERNAME): str,
-                    vol.Required(CONF_PASSWORD): str,
-                    vol.Optional(CONF_VENDOR, default="neato"): vol.In(VALID_VENDORS),
-                }
-            ),
-            description_placeholders={"docs_url": DOCS_URL},
-            errors=errors,
-        )
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
 
-    async def async_step_import(self, user_input):
-        """Import a config flow from configuration."""
-
-        if self._async_current_entries():
+    async def async_step_user(self, user_input=None) -> dict:
+        """Create an entry for the flow."""
+        current_entries = self._async_current_entries()
+        if current_entries and current_entries[0].version == self.VERSION:
             return self.async_abort(reason="already_configured")
 
-        username = user_input[CONF_USERNAME]
-        password = user_input[CONF_PASSWORD]
-        vendor = user_input[CONF_VENDOR]
-
-        error = await self.hass.async_add_executor_job(
-            self.try_login, username, password, vendor
-        )
-        if error is not None:
-            _LOGGER.error(error)
-            return self.async_abort(reason=error)
+        return await super().async_step_user(user_input=user_input)
 
-        return self.async_create_entry(
-            title=f"{username} (from configuration)",
-            data={
-                CONF_USERNAME: username,
-                CONF_PASSWORD: password,
-                CONF_VENDOR: vendor,
-            },
-        )
+    async def async_step_reauth(self, config_entry: ConfigEntry) -> dict:
+        """Perform reauth upon migration of old entries."""
+        # self.hass.config_entries.async_remove(config_entry.unique_id)
+        return self.async_show_form(step_id="user")
 
-    @staticmethod
-    def try_login(username, password, vendor):
-        """Try logging in to device and return any errors."""
-        this_vendor = None
-        if vendor == "vorwerk":
-            this_vendor = Vorwerk()
-        else:  # Neato
-            this_vendor = Neato()
-
-        try:
-            Account(username, password, this_vendor)
-        except NeatoLoginException:
-            return "invalid_auth"
-        except NeatoRobotException:
-            return "unknown"
-
-        return None
+    async def async_oauth_create_entry(self, data: dict) -> dict:
+        """Create an entry for the flow. Update an entry if one already exist."""
+        current_entries = self._async_current_entries()
+        if current_entries and current_entries[0].version == 1:
+            self.hass.config_entries.async_update_entry(
+                current_entries[0], title=self.flow_impl.name, data=data
+            )
+            # TODO: how to update the entry version?
```

Comment:
We don't have an interface for changing version yet. Just set the new version directly on the entry.

Reply:
Done